src/cart/crt_hg.c

```diff
@@ -402,117 +434,113 @@ crt_hg_log(FILE *stream, const char *fmt, ...)
 
 /* to be called only in crt_init */
 int
-crt_hg_init(crt_phy_addr_t *addr, bool server)
+crt_hg_init(void)
 {
-	char			*info_string = NULL;
-	struct crt_hg_gdata	*hg_gdata = NULL;
-	hg_class_t		*hg_class = NULL;
-	struct hg_init_info	 init_info = HG_INIT_INFO_INITIALIZER;
-	int			 rc = 0;
+	int	rc = 0;
 
 	if (crt_initialized()) {
 		D_ERROR("CaRT already initialized.\n");
 		D_GOTO(out, rc = -DER_ALREADY);
 	}
 
+	#define EXT_FAC DD_FAC(external)
+
 	/* import HG log */
 	hg_log_set_func(crt_hg_log);
-	hg_log_set_stream_debug((FILE *)(intptr_t)(D_LOGFAC | DLOG_DBG));
-	hg_log_set_stream_warning((FILE *)(intptr_t)(D_LOGFAC | DLOG_WARN));
-	hg_log_set_stream_error((FILE *)(intptr_t)(D_LOGFAC | DLOG_ERR));
+	hg_log_set_stream_debug((FILE *)(intptr_t)(EXT_FAC | DLOG_DBG));
+	hg_log_set_stream_warning((FILE *)(intptr_t)(EXT_FAC | DLOG_WARN));
+	hg_log_set_stream_error((FILE *)(intptr_t)(EXT_FAC | DLOG_ERR));
 
-	D_ASSERTF(*addr == NULL, "Can only be called in crt_init().\n");
+	#undef EXT_FAC
+out:
+	return rc;
+}
 
-	rc = crt_get_info_string(&info_string);
+/* Shared HG class, used in SEP mode */
+static hg_class_t *sep_hg_class;
```

Comment:
correct, this will also be addressed during multiprovider support along with other changes to make every setting per-provider instead of currently using 1 global setting that defines current provider.